lib/Runtime/Library/TypedArray.cpp

```diff
@@ -121,9 +159,9 @@ namespace Js
                         Var lengthVar = JavascriptOperators::OP_GetProperty(jsArraySource, PropertyIds::length, scriptContext);
                         if (JavascriptOperators::GetTypeId(lengthVar) == TypeIds_Undefined)
                         {
-                            JavascriptError::ThrowTypeError(
-                                scriptContext, JSERR_InvalidTypedArray_Constructor);
+                            return CreateNewInstanceFromObj(jsArraySource, scriptContext, elementSize, pfnCreateTypedArray);
```

Comment:
Should we check for JavascriptOperators::IsIterable to call from CreateNewInstanceFromObj?